backend/data.go

```diff
@@ -24,10 +25,17 @@ type DataQuery struct {
 	JSON          json.RawMessage
 }
 
-// QueryDataResponse holds the results for a given query.
+// QueryDataResponse contains the results from a QueryDataRequest.
 type QueryDataResponse struct {
-	Frames   []*data.Frame
-	Metadata map[string]string
+	Responses map[string]*DataResponse
+	Metadata  map[string]string
+}
+
+// DataResponse contains the results frim a DataQuery.
+type DataResponse struct {
+	Frames []*data.Frame
+	Meta   json.RawMessage
```

Comment:
is this meta object all "custom" fields?  

Where do we expect them to end up?

Reply:
All custom as far as I can think.

I'm not sure where we expect them to end up. What I expect it to be is any metadata about the query. So for example, many client APIs will return some metadata when you execute a query (bytes read for example). I would expect that to live here, since it is metadata about the query response.